jvector-base/src/main/java/io/github/jbellis/jvector/disk/GraphCache.java

```diff
@@ -65,12 +64,12 @@ public long ramBytesUsed()
     private static final class HMGraphCache extends GraphCache
     {
         // Map is created on construction and never modified
-        private final Map<Integer, CachedNode> cache;
+        private final Int2ObjectHashMap<CachedNode> cache;
         private long ramBytesUsed = 0;
 
         public HMGraphCache(GraphIndex<float[]> graph, int distance) {
             try (var view = graph.getView()) {
-                HashMap<Integer, CachedNode> tmpCache = new HashMap<>();
+                var tmpCache = new Int2ObjectHashMap<CachedNode>();
                 cacheNeighborsOf(tmpCache, view, view.entryNode(), distance);
                 // Assigning to a final value ensure it is safely published
                 cache = tmpCache;
```

Comment:
It's probably a premature optimization, but when I looked at this the other day, I noticed that we could technically rebuild the map at this point with the correct size since we know `tmpCache`'s size, which we only get after walking the graph IIUC. Now that we're using the primitive map, I think it takes up more space to have unused elements, but I'm not sure how much it affects performance.

Probably not worth a change... but I wanted to mention it in case we find it takes up more space than necessary.